oshi-core/src/main/java/oshi/software/os/linux/LinuxUserInfo.java

```diff
@@ -0,0 +1,37 @@
+package oshi.software.os.linux;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```

Comment:
Instead of a `Map<String, String>`, is not preferrable an object that return the username and the userid? For example:

```
public class User {
    private String name;
    private int id;

    public User(String name, int id) {
        this.name = name;
        this.id = id;
    }
    public String getName() {
        return this.name;
    }

    public int getId() {
        return this.id;
    }
}
```

Reply:
If we wanted to expose the user/group database to the user I'd agree. But as I read the code, this is just an internally cached lookup of UID->user and GID->group, and a `HashMap` is the most efficient way to store that lookup.  Otherwise you'd have to iterate through all your `User` objects looking for a matching `User.id` to match your `uid`.

Reply:
Yes, for now this is just a lookup. I have created issue (https://github.com/oshi/oshi/issues/434) to work on more the more general UserInformation feature.

Reply:
Good that you're thinking ahead.  Whatever class you design for the lookup, perhaps you might think about the structure you'll have for #434 and whether it makes sense to build the skeleton now.